packages/synapse-react-client/src/mocks/msw/handlers/entityHandlers.ts

```diff
@@ -231,11 +237,18 @@ export const getEntityHandlers = (backendOrigin: string) => [
       }
 
       const referenceList = req.body as { references: Reference[] }
-      const entityData = mockEntities
-        .filter(entity =>
-          referenceList.references.find(ref => ref.targetId === entity.id),
-        )
-        .map(entity => entity.entityHeader)
+      const entityData: EntityHeader[] = referenceList.references
+        .map(ref => {
+          const entityHeader = getMatchingMockEntity(ref.targetId)?.entityHeader
+          if (entityHeader) {
+            return {
+              ...entityHeader,
+              // ID should match what is passed in (i.e. may or may not have syn prefix)
+              id: ref.targetId,
```

Comment:
Match behavior of the backend - if the client asks for `syn123`, the returned ID is `syn123`. If the client asks for `123`, the returned ID is `123`.